pkgs/os-specific/linux/mdadm/default.nix

```diff
@@ -12,15 +12,12 @@ stdenv.mkDerivation rec {
 
   makeFlags = [
     "NIXOS=1" "INSTALL=install" "BINDIR=$(out)/sbin"
-    "SYSTEMD_DIR=$(out)/lib/systemd/system"
```

Comment:
Yeah. Unfortunately the original change was kind of opaque and didn't have any tests either - so I'm not really sure whether those are doing anything helpful anyway ... (see below)

Reply:
@fmap Would you be available to comment on this topic, as you added this line? My guess would be it might be useful for running Nixpkgs' mdadm on a non-NixOS machine, but not using one myself I couldn't say for sure.

Reply:
I really think we should leave this in and then override the upstream units where necessary. I have already pointed out in another comment how this can be achieved... Let me know if any help is needed.